libraries/chain/steem_evaluator.cpp

```diff
@@ -1936,7 +1956,30 @@ void hf20_vote_evaluator( const vote_operation& o, database& _db )
 
       _db.modify( voter, [&]( account_object& a )
       {
-         a.voting_manabar.use_mana( used_mana.to_uint64() );
+         if( _db.has_hardfork( STEEM_HARDFORK_0_21__3336 ) && dgpo.downvote_pool_percent > 0 && o.weight < 0 )
+         {
+            if( used_mana.to_uint64() > a.downvote_manabar.current_mana )
+            {
+               /* used mana is always less than downvote_mana + voting_mana because the amount used
+                * is a fraction of max( downvote_mana, voting_mana ). If more mana is consumed than
```

Comment:
This comment doesn't appear to be true since used mana is taking a fraction of max(4*dvmana, mana) under current parameters. Is there an issue when someone drains their regular mana and then starts draining the DV mana?  Or is that fraction guaranteed to be < 1/4? (Or dvpoolfraction).

Reply:
Currently, used mana ends up being 1/50th of the remaining voting mana.

The downvote logic uses up 1/50th of the remaining voting mana or 4/50th of the remaining downvote mana, whichever is greater (from downvote mana, then upvote mana). If 1/50th of the remaining voting mana is larger than the remaining downvote mana, the the remainder is taken from the voting mana. In all cases, the amount of used mana must always be less than the remaining voting mana because it is 1/50th of the remaining mana.

In the other scenario, where downvote mana is high and voting mana is low, the downvote mana will not be over consumed because the max that can be used is 4/50th. Once downvote mana is consumed low enough that upvote mana is being used, then 1/50th of the voting mana is used.

Under no scenario can the amount of used mana be more than voting mana + downvote mana because used mana is always a fraction of remaining mana.